contracts/vaults/TreasuryVault/RibbonAutocallVault.sol

```diff
@@ -208,25 +208,27 @@ contract RibbonAutocallVault is RibbonTreasuryVaultLite, AutocallVaultStorage {
         uint256 strikePrice =
             currentOption == address(0) ? 0 : currentOToken.strikePrice();
 
-        (uint256 autocallTS, uint256 nCBBreaches, uint256 lastCBBreach) =
-            _autocallState(expiry);
-
-        // If before expiry, attempt to autocall
-        if (block.timestamp < expiry) {
-            // Require autocall barrier hit at least once
-            require(autocallTS < block.timestamp, "!autocall");
-            // Burn the unexpired oTokens
-            _burnRemainingOTokens();
-            // Require vault possessed all oTokens sold to counterparties
-            require(vaultState.lockedAmount == 0, "!withdrawnCollateral");
-        }
-
-        // Calculate coupons earned
-        (, , uint256 returnAmt) = _couponsEarned(nCBBreaches, lastCBBreach);
+        if (currentOption != address(0)) {
```

Comment:
can we remove `if (currentOption != address(0)) {`

I don't think we need it after adding ` if (lastCBBreach > 0) {`

Reply:
what happens when we remove `if (currentOption != address(0)) {` is that on the 1st roll to next round which is when "currentOption == address(0)" it reverts with `!obsPrice` since it is trying to get oracle price at timestamp zero